### PR TITLE
[pack] Include py nuget

### PR DESCRIPTION
**Pull request by maiqbal11**
Fixes https://github.com/Azure/azure-functions-host/issues/4935. Part 2 of https://github.com/Azure/azure-functions-host/pull/4985.

### CODE REVIEW

**Comment by maiqbal11** [comment]
@pragnagopa fyi. I pulled in the changes from https://github.com/Azure/azure-functions-host/pull/4985 and added the NuGet from https://github.com/Azure/azure-functions-python-worker/pull/555. Works with Python 3.7 E2E. 

Will rework this PR once we close on https://github.com/Azure/azure-functions-host/pull/4985.

File: run-tests.ps1 [hunk 1]
```diff
@@ -30,21 +30,7 @@ function RunTest([string] $project, [string] $description,[bool] $skipBuild = $f
 
 
 $tests = @(
-  @{project = "..\tools\ExtensionsMetadataGenerator\test\ExtensionsMetadataGeneratorTests"; description="ExtensionsMetadataGenerator Tests"},
-  @{project ="WebJobs.Script.Tests"; description="Unit Tests"},
```
Comment (pragnagopa) [inline run-tests.ps1]: undo

File: build/python.props [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<Project>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Functions.PythonWorkerRunEnvironments" Version="1.0.20191003.2" />
```
Comment (pragnagopa) [inline build/python.props]: Microsoft.Azure.Functions.PythonWorkerRunEnvironments - is this the name of nuget package? why not just Microsoft.Azure.Functions.PythonWorker?

Comment (pragnagopa) [reply build/python.props]: Version does not look right. Should it be: 1.0.20191003?

File: run-tests.ps1 (hunk 1 above)
Comment (maiqbal11) [reply run-tests.ps1]: Will do. Had just wanted to keep runs short while validating.

File: build/python.props (hunk 2 above)
Comment (maiqbal11) [reply build/python.props]: > Microsoft.Azure.Functions.PythonWorkerRunEnvironments - is this the name of nuget package? why not just Microsoft.Azure.Functions.PythonWorker?

This was the previous name when we were packaging directly with the core tools. I'll change it to Microsoft.Azure.Functions.PythonWorker.

Comment (maiqbal11) [reply build/python.props]: > Version does not look right. Should it be: 1.0.20191003?

The version is correct. The `.2` at the end is meant to account for the build number for that particular day. For example, if you do multiple builds in a day, the `.x` keeps the build number unique.

Comment (pragnagopa) [reply build/python.props]: We should use valid Semantic versioning as described here: https://semver.org. Need to fix the build to produce a valid version

Comment (maiqbal11) [reply build/python.props]: > We should use valid Semantic versioning as described here: https://semver.org. Need to fix the build to produce a valid version

Will modify it to remove the `.` after the date so that it still remains unique but follows correct patch version syntax.

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestFixture.cs [hunk 3]
```diff
@@ -98,6 +99,10 @@ public async Task InitializeAsync()
                 Environment.SetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName, _functionsWorkerRuntime);
                 Environment.SetEnvironmentVariable(LanguageWorkerConstants.FunctionsWorkerProcessCountSettingName, _workerProcessCount.ToString());
             }
+            if (!string.IsNullOrEmpty(_functionsWorkerRuntimeVersion))
+            {
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestFixture.cs]: Reset this variable in DisposeAsync

File: build/python.props [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<Project>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Functions.PythonWorker" Version="1.0.201910045" />
```
Comment (pragnagopa) [inline build/python.props]: can you use build number instead of date?

Comment (maiqbal11) [reply build/python.props]: That is currently how I've defined the build number to be unique. Don't think there should be any issues with using this.

Comment (pragnagopa) [reply build/python.props]: across build systems we use build number generated by the CI system. Would be good to align versions across components and repos

Comment (maiqbal11) [reply build/python.props]: This **is** the build number generated by Azure DevOps by default. They use the current date. To make it unique, you can add a `rev` version in front of it, which is what I've done. Any different option in Azure DevOps at least would require manually incrementing patch versions.

Comment (pragnagopa) [reply build/python.props]: We have other repos : core tools, extensions using devops. Adding @soninaren / @ankitkumarr  to point to how we generate version for nuget packages

Comment (maiqbal11) [reply build/python.props]: I'm not sure if we need to strive for consistency here. This is also a valid way of incrementing the patch number.

Comment (maiqbal11) [reply build/python.props]: Node versions are currently hard coded and don't rely on the Azure DevOps pipeline build numbers, which is also ok: https://github.com/Azure/azure-functions-nodejs-worker/blob/dev/azure-pipelines.yml.

**Comment by maiqbal11** [comment]
Adding `do not merge` label as the changes in https://github.com/Azure/azure-functions-python-worker/pull/556 need to included in the final NuGet packaged with the Host.

**Comment by ankitkumarr** [comment]
I know we want the Python worker in the Windows host for core-tools. But, does this mean that our Windows platform in Azure will also have python workers (as they don't really need that)?

**Comment by maiqbal11** [comment]
> I know we want the Python worker in the Windows host for core-tools. But, does this mean that our Windows platform in Azure will also have python workers (as they don't really need that)?

@ankitkumarr, I've added code that removes the python worker from the private site extensions. It should not be there in the Windows platform.

**Comment by maiqbal11** [comment]
@pragnagopa, this should be good to go!

File: build.ps1 [hunk 5]
```diff
@@ -233,6 +233,7 @@ function cleanExtension([string] $bitness) {
     Remove-Item -Recurse -Force "$privateSiteExtensionPath\$bitness\publish" -ErrorAction SilentlyContinue
     Remove-Item -Recurse -Force "$privateSiteExtensionPath\$bitness\runtimes\linux" -ErrorAction SilentlyContinue
     Remove-Item -Recurse -Force "$privateSiteExtensionPath\$bitness\runtimes\osx" -ErrorAction SilentlyContinue
+    Remove-Item -Recurse -Force "$privateSiteExtensionPath\$bitness\workers\python" -ErrorAction SilentlyContinue
```
Comment (pragnagopa) [inline build.ps1]: @alrod / @fabiocav - can you please sign off on build changes?

File: src/WebJobs.Script/OutOfProc/Rpc/Configuration/WorkerConfigFactory.cs [hunk 6]
```diff
@@ -130,13 +130,12 @@ internal void AddProvider(string workerDir)
...
                 GetDefaultExecutablePathFromAppSettings(workerDescription, languageSection);
                 AddArgumentsFromAppSettings(workerDescription, languageSection);
-                workerDescription.ApplyDefaultsAndValidate();
-                _workerDescripionDictionary[workerDescription.Language] = workerDescription;
-
                 if (IsHydrationNeeded(workerDescription.DefaultWorkerPath))
                 {
                     workerDescription.DefaultWorkerPath = GetHydratedWorkerPath(workerDescription);
                 }
+                workerDescription.ApplyDefaultsAndValidate();
```
Comment (fabiocav) [inline src/WebJobs.Script/OutOfProc/Rpc/Configuration/WorkerConfigFactory.cs]: This is a bit out of scope for this change. Is this required in order to get your tests to work? If so, we might want to consider moving those tests to a different PR or at least keeping them in a different commit

Comment (maiqbal11) [reply src/WebJobs.Script/OutOfProc/Rpc/Configuration/WorkerConfigFactory.cs]: Yes, it's needed for the tests. We need to populated the default worker path with the proper values for the worker file to be found.

File: appveyor.yml [hunk 7]
```diff
@@ -42,6 +42,8 @@ for:
   - ps: >-
       $env:CommitHash = "$env:APPVEYOR_REPO_COMMIT"
 
+      $env:Path="C:\Python37-x64;$env:Path"
```
[the remark this answers is not in the document]
Comment (maiqbal11) [reply appveyor.yml]: I should be able to define it as part of the AppVeyor configuration. Will take a look.

[the remark this answers is not in the document]
Comment (maiqbal11) [reply appveyor.yml]: I've defined a separate `PythonVersionPath` environment variable to contain the `C:\Python37-x64` portion. When searching for recommended ways of consuming different Python versions, the path addition was the recommended approach: https://help.appveyor.com/discussions/questions/23642-how-do-i-have-my-python-script-run-with-a-specific-version-of-python.

File: build.ps1 (hunk 5 above)
Comment (alrod) [reply build.ps1]: Signed oof.

**Comment by maiqbal11** [comment]
@fabiocav, @pragnagopa, this is ready for check in. Could someone please approve? Thanks!
